test/test_function_schema.py

```diff
@@ -10,8 +11,81 @@ def test_serialize_and_deserialize(self):
         # so far we have around 1700 registered schemas
         self.assertGreater(len(schemas), 1000)
         for schema in schemas:
-            parsed_schema = torch._C.parse_schema(str(schema))
+            parsed_schema = parse_schema(str(schema))
             self.assertEqual(parsed_schema, schema)
+            self.assertTrue(parsed_schema.is_backcompat_with(schema))
+
+    def test_backward_compatible_optional_arg(self):
+        old_schema = parse_schema('any(Tensor self, int dim, bool keepdim=False) -> Tensor')
+        new_schema = parse_schema('any(Tensor self, int? dim, bool keepdim=False) -> Tensor')
+        self.assertTrue(new_schema.is_backcompat_with(old_schema))
+        self.assertFalse(old_schema.is_backcompat_with(new_schema))
+
+    def test_backward_compatible_default_value(self):
+        old_schema = parse_schema('any(Tensor self, int dim, bool keepdim=False) -> Tensor')
+        new_schema = parse_schema('any(Tensor self, int dim=5, bool keepdim=False) -> Tensor')
+        self.assertTrue(new_schema.is_backcompat_with(old_schema))
+        self.assertFalse(old_schema.is_backcompat_with(new_schema))
+
+    def test_backward_compatible_add_args(self):
+        old_schema = parse_schema('any(Tensor self, int dim) -> Tensor')
+        new_schema = parse_schema('any(Tensor self, int dim, bool keepdim=False) -> Tensor')
+        self.assertTrue(new_schema.is_backcompat_with(old_schema))
+        self.assertFalse(old_schema.is_backcompat_with(new_schema))
+        new_schema = parse_schema('any(Tensor self, int dim, bool? keepdim) -> Tensor')
```

Comment:
This _isn't_ backward compatible. T? doesn't mean that an argument goes unspecified, it means that the caller may pass None or T

Reply:
good catch, fixed.